.github/workflows/checks.yml

```diff
@@ -4,21 +4,24 @@ on: [push, pull_request]
 
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
 
     steps:
-      - name: Install Indent
+      - name: Install Dependencies
         run: |
           sudo apt-get -qq update
-          sudo apt-get install -y indent
+          # sudo apt-get install -y clang-format
 
       - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
 
       - name: Shell script static analysis
-        run: shellcheck bin/fetch-configlet bin/verify-indent bin/check-unitybegin bin/run-tests indent.sh
+        run: shellcheck bin/fetch-configlet bin/verify-unity-version bin/check-unitybegin bin/run-tests format.sh
```

Comment:
Does `format.sh` have a non-zero exit code when the formatter needs to make a change?

Reply:
When checking if a change is needed, as in `checks.yaml`, the `--dry-run` argument is used. This results in a non-zero (`123`) exist code if incorrect formatting is detected. Which should then cause the GitHub workflow to fail, as expected.

When running `format.sh` to actually change the code to be correctly formatted, the `-i` (in-place edit) argument is used instead. If nay changes are needed clang-format makes them and the exit code is `0`. I think this is also as expected. Do you agree or do I misunderstand?

```console
# Per the checks.yaml used in CI
~/c$ find exercises/practice/ -path '*/test-framework' -prune -o \
>        \( -iname '*.h' -o -iname '*.c' \) -print0 |\
>        xargs --null clang-format --dry-run --Werror 
exercises/practice/diamond/test_diamond.c:37:13: error: code should be clang-formatted [-Wclang-format-violations]
      " A ",
            ^
exercises/practice/diamond/test_diamond.c:38:13: error: code should be clang-formatted [-Wclang-format-violations]
      "B B",
            ^
~/c$ echo $?
123       # i.e. incorrect formatting was detected as shown in the above output by the clang-format

# Per the format.sh script for use by contributors
~/c$ find exercises/practice/ -path 'exercises/practice/*/test-framework' -prune -o \
>        \( -iname '*.h' -o -iname '*.c' \) -print0 |\
>        xargs --null clang-format -i
~/c$ echo $?
0        # i.e. any needed formatting changes were correctly made
```

Reply:
Sorry, I totally missed that this was the `shellcheck` step and somehow convinced myself this was the list of scripts to run.